libcudacxx/test/libcudacxx/cuda/for_each_canceled/for_each_canceled.pass.cpp

```diff
@@ -0,0 +1,220 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of libcu++, the C++ Standard Library for your entire system,
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+// UNSUPPORTED: pre-sm-100
+// UNSUPPORTED: !nvcc
+// UNSUPPORTED: nvrtc
+// UNSUPPORTED: c++98, c++03, c++14, c++17
+
+#include <cuda/for_each_canceled>
+
+__device__ void vec_add_impl1(int* a, int* b, int* c, int n, dim3 block_idx)
+{
+  int idx = threadIdx.x + block_idx.x * blockDim.x;
+  if (idx < n)
+  {
+    c[idx] += a[idx] + b[idx];
+  }
+}
+
+__device__ void vec_add_impl2(int* a, int* b, int* c, int n, dim3 block_idx)
+{
+  int x_idx = threadIdx.x + (block_idx.x * blockDim.x);
+  int y_idx = threadIdx.y + (block_idx.y * blockDim.y);
+  int idx   = x_idx + y_idx * (blockDim.x * gridDim.x);
+  if (idx < n)
+  {
+    c[idx] += a[idx] + b[idx];
+  }
+}
+
+__device__ void vec_add_impl3(int* a, int* b, int* c, int n, dim3 block_idx)
+{
+  int x_idx = threadIdx.x + (block_idx.x * blockDim.x);
+  int y_idx = threadIdx.y + (block_idx.y * blockDim.y);
+  int z_idx = threadIdx.z + (block_idx.z * blockDim.z);
+  int idx   = x_idx + y_idx * (blockDim.x * gridDim.x) + z_idx * (blockDim.x * gridDim.x * blockDim.y * gridDim.y);
+  if (idx < n)
+  {
+    c[idx] += a[idx] + b[idx];
+  }
+}
+
+__global__ void vec_add_det1(int* a, int* b, int* c, int n, int tidx = 0)
+{
+  cuda::__detail::__for_each_canceled_block<1>(threadIdx.x == tidx, [=](dim3 block_idx) {
+    vec_add_impl1(a, b, c, n, block_idx);
+  });
+}
+
+__global__ void vec_add_det2(int* a, int* b, int* c, int n, int tidx = 0)
+{
+  cuda::__detail::__for_each_canceled_block<2>(threadIdx.x == tidx && threadIdx.y == tidx, [=](dim3 block_idx) {
+    vec_add_impl2(a, b, c, n, block_idx);
+  });
+}
+
+__global__ void vec_add_det3(int* a, int* b, int* c, int n, int tidx = 0)
+{
+  cuda::__detail::__for_each_canceled_block<3>(
+    threadIdx.x == tidx && threadIdx.y == tidx && threadIdx.z == tidx, [=](dim3 block_idx) {
+      vec_add_impl3(a, b, c, n, block_idx);
+    });
+}
+
+__global__ void vec_add1(int* a, int* b, int* c, int n, int tidx = 0)
+{
+  cuda::for_each_canceled_block<1>([=](dim3 block_idx) {
+    vec_add_impl1(a, b, c, n, block_idx);
+  });
+}
```

Comment:
Please remove this unused parameter:
```suggestion
__global__ void vec_add1(int* a, int* b, int* c, int n)
{
  cuda::for_each_canceled_block<1>([=](dim3 block_idx) {
    vec_add_impl1(a, b, c, n, block_idx);
  });
}
```
Applies to the following two kernels as well.